app/ante/evm/fee_checker.go

```diff
@@ -109,9 +109,15 @@ func NewDynamicFeeChecker(k dynamicFeeEVMKeeper) authante.TxFeeChecker {
 // unit of gas is fixed and set by each validator, and the tx priority is computed from the gas price.
 func checkTxFeeWithValidatorMinGasPrices(ctx sdk.Context, tx sdk.FeeTx) (sdk.Coins, int64, error) {
 	feeCoins := tx.GetFee()
-	gas := tx.GetGas()
 	minGasPrices := ctx.MinGasPrices()
 
+	// Check tx gas is not bigger than max.MaxInt64.
+	// if it is, then charge the max
+	gas := int64(math.MaxInt64)
+	if g := tx.GetGas(); g < math.MaxInt64 {
```

Comment:
It's better if you add the check to the MsgEthereumTx `ValidateBasic()`

Reply:
Will do that - also plan to use Int too to do a proper check